### PR TITLE
Added support for extension fee-0.7

### PR DESCRIPTION
**Pull request by wbond**

### CODE REVIEW

File: check.go [hunk 1]
```diff
@@ -56,15 +56,16 @@ func encodeDomainCheck(buf *bytes.Buffer, domains []string, greeting Greeting) e
...
-		if supportsFee05 {
-			feeURN = ExtFee05
-		}
+	var feeURN string
+	if greeting.SupportsExtension(ExtFee05) {
+		feeURN = ExtFee05
+	} else if greeting.SupportsExtension(ExtFee06) {
+		feeURN = ExtFee06
+	} else if greeting.SupportsExtension(ExtFee07) {
+		feeURN = ExtFee07
```
Comment (cee-dub) [inline check.go]: Maybe time for a `switch` statement?

**Comment by case** [comment]
@cee-dub this look good to go?

File: check.go [hunk 2]
```diff
@@ -56,15 +56,17 @@ func encodeDomainCheck(buf *bytes.Buffer, domains []string, greeting Greeting) e
...
-		}
+	var feeURN string
+	switch {
+	case greeting.SupportsExtension(ExtFee05):
+		feeURN = ExtFee05
+	case greeting.SupportsExtension(ExtFee06):
+		feeURN = ExtFee06
+	case greeting.SupportsExtension(ExtFee07):
+		feeURN = ExtFee07
+	}
```
Comment (cee-dub) [inline check.go]: This appears to let the lowest version number win, is that correct?

Comment (wbond) [inline check.go]: Yes, because fee-0.5 has a simple attribute `premium="1"` whereas later versions require matching on class name.

**Comment by cee-dub** [comment]
LGTM

**Pull request merged**